src/main/java/com/greenfox/tribes/game/controllers/Work.java

```diff
@@ -36,4 +37,52 @@ public String logWork(@RequestParam("id") long id) {
     activityService.logActivity(ActivityType.WORK, id);
     return "redirect:/activity/work";
   }
+
+  @GetMapping("/pvp")
+  public String pvp(Model model) {
+    /*    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    WastelandUser user = userRepository.findByUsername(auth.getName()).get();
+    Persona[] combatants = activityService.arenaFight(user.getPersona().getId());
+    //model.addAttribute("combatants", combatants);
+    System.out.println(combatants[0].getCharacterName() + " vs. " + combatants[1].getCharacterName());
+    if (activityService.isFinished(user.getPersona().getId())) {
+      activityService.decideFightResult(combatants);
+    }else{
+
+    }*/
+
+    return "game-sites/work";
+  }
+
+  @GetMapping("/pvp/done")
+  public String pvpDone(
+      Model model, @RequestParam("userId") long id, @RequestParam("enemyId") long enemyId) {
+
+    Persona user = userRepository.findById(id).get().getPersona();
+    Persona enemy = userRepository.findById(enemyId).get().getPersona();
+    Persona[] combatants = {user, enemy};
+    // model.addAttribute("combatants", combatants);
+    System.out.println(
+        combatants[0].getCharacterName() + " vs. " + combatants[1].getCharacterName());
+    if (activityService.isFinished(user.getId())) {
+      activityService.decideFightResult(combatants);
+    }
+
+    return "game-sites/work";
+  }
+
+  @GetMapping("/pvp/log")
+  public String logPvp(@RequestParam("id") long id) {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    WastelandUser user = userRepository.findByUsername(auth.getName()).get();
+    Persona[] combatants = activityService.arenaFight(user.getPersona().getId());
+    activityService.logActivity(ActivityType.PVP, id);
+    String adress =
+        "redirect:/activity/pvp/done"

```

Comment:
It would be easier to user RedirectAttributes and use addAttribute (or something like that) so that you don't have to do this:
+ "?userId="
            + combatants[0].getPlayer().getId()
            + "&enemyId="
            + combatants[1].getPlayer().getId();
  
You don't have to change this, just a tip for next time